utils.c

```diff
@@ -35,17 +35,56 @@ void xfree(void* ptr) {
 #   define INPUT_KEEPS_COPY
 #endif
 
+#if defined INPUT_KEEPS_COPY
+#    define is_copy_flag(flag)
```

Comment:
I'd remove `is_copy_flag` and simply use `ifdef`, as the macro is used just in one function